cmd/node/node.go

```diff
@@ -689,20 +695,30 @@ func (app *SpacemeshApp) startAPIServices(postClient api.PostAPI, net api.Networ
 	// gateway server is enabled without enabling at least one GRPC service.
 
 	// Make sure we only start the server once
-	startService := func(svc grpcserver.ServiceAPI) {
+	name := app.nodeID.ShortString()
+	lg := log.NewDefault(name).WithFields(app.nodeID)
+
+	startService := func(svc server.API) {
 		if app.newgrpcAPIService == nil {
-			app.newgrpcAPIService = grpcserver.NewServer(apiConf.NewGrpcServerPort)
+			app.newgrpcAPIService = server.New(apiConf.NewGrpcServerPort, server.WithLogger(
+				app.addLogger(GrpcLogger, lg),
+			))
 			app.newgrpcAPIService.Start()
 		}
-		svc.RegisterService(app.newgrpcAPIService)
+		svc.Register(app.newgrpcAPIService)
 	}
 
 	// Start the requested services one by one
 	if apiConf.StartNodeService {
-		startService(grpcserver.NewNodeService(net, app.mesh, app.clock, app.syncer))
+		startService(servicenode.New(net, app.mesh, app.clock, app.syncer, servicenode.WithLogger(
+			app.addLogger(NodeLogger, lg),
+		)))
 	}
 	if apiConf.StartMeshService {
-		startService(grpcserver.NewMeshService(net, app.mesh, app.clock, app.syncer))
+		lvl := app.loggers[MeshLogger] // because we already set the logger in the map
+		startService(servicemesh.New(net, app.mesh, app.clock, app.syncer, servicemesh.WithLogger(
+			lg.SetLevel(lvl).WithName(MeshLogger),
```

Comment:
didn't you mean to use `addLogger` here as well?

Reply:
I've noticed that meshlogger it's already set in the map so I figured I can just get ptr from the map and just pass/construct only the outer required thing. I can change to addLogger but it will overwrite the same mesh logger or we could add a new var key with a different name. You're call.

Reply:
These are definitely two different things - `MeshLogger` should be distinct from `MeshServiceLogger`, which is what you want here. The GRPC services should get distinct loggers with the name `<Svcname>ServiceLogger`.

Reply:
I did used/updated all naming logger acordignly.